src/main/java/org/openrewrite/staticanalysis/EqualsAvoidsNull.java

```diff
@@ -76,104 +76,110 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
                 new JavaVisitor<ExecutionContext>() {
                     @Override
                     public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
-                        J.MethodInvocation m = (J.MethodInvocation) super.visitMethodInvocation(method, ctx);
+                        return checkLiteralsFirstInComparisons((J.MethodInvocation) super.visitMethodInvocation(method, ctx));
+                    }
 
-                        if (!isStringComparisonMethod(m) || !hasCompatibleArgument(m) || m.getSelect() instanceof J.Literal) {
-                            return m;
+                    private J checkLiteralsFirstInComparisons(J.MethodInvocation m) {
+                        if (isStringComparisonMethod(m) &&
+                                hasCompatibleArgument(m) &&
+                                !(m.getSelect() instanceof J.Literal)) {
+                            return applyLiteralsFirstInComparisons(m, m.getArguments().get(0));
                         }
+                        return m;
+                    }
 
-                        maybeHandleParentBinary(m, getCursor().getParentTreeCursor().getValue());
-                        Expression firstArgument = m.getArguments().get(0);
-
+                    private J applyLiteralsFirstInComparisons(J.MethodInvocation m, Expression firstArgument) {
+                        checkPotentialNullCheck(m, getCursor().getParentTreeCursor().getValue());
```

Comment:
I don't know how `checkPotentialNullCheck` is working, but from the storytelling here, I would expect this to act as an early exit. A null check does not get flipped, which doesn’t make sense because null is checked again below in `literalsFirstInComparisonsNull`, so it must be treated as well.  

Therefore, it seems more like a treatment or preparation step and should be renamed to `preparePotentialNullCheck` to better reflect its purpose in allowing the process to continue.